v2/api/devices/v1api20210702/structure.txt

```diff
@@ -0,0 +1,841 @@
+│   │   │   │   │   ├── "RegistryRead, ServiceConnect, DeviceConnect"
+│   │   │   │   │   ├── "RegistryWrite"
+│   │   │   │   │   ├── "RegistryWrite, DeviceConnect"
+│   │   │   │   │   ├── "RegistryWrite, ServiceConnect"
+│   │   │   │   │   ├── "RegistryWrite, ServiceConnect, DeviceConnect"
+│   │   │   │   │   ├── "ServiceConnect"
+│   │   │   │   │   └── "ServiceConnect, DeviceConnect"
+│   │   │   │   └── SecondaryKey: *string
+│   │   │   ├── CloudToDevice: *Object (3 properties)
+│   │   │   │   ├── DefaultTtlAsIso8601: *string
+│   │   │   │   ├── Feedback: *Object (3 properties)
+│   │   │   │   │   ├── LockDurationAsIso8601: *string
+│   │   │   │   │   ├── MaxDeliveryCount: Validated<*int> (2 rules)
+│   │   │   │   │   │   ├── Rule 0: Maximum: 100
+│   │   │   │   │   │   └── Rule 1: Minimum: 1
+│   │   │   │   │   └── TtlAsIso8601: *string
+│   │   │   │   └── MaxDeliveryCount: Validated<*int> (2 rules)
+│   │   │   │       ├── Rule 0: Maximum: 100
+│   │   │   │       └── Rule 1: Minimum: 1
+│   │   │   ├── Comments: *string
+│   │   │   ├── DisableDeviceSAS: *bool
+│   │   │   ├── DisableLocalAuth: *bool
+│   │   │   ├── DisableModuleSAS: *bool
+│   │   │   ├── EnableDataResidency: *bool
+│   │   │   ├── EnableFileUploadNotifications: *bool
+│   │   │   ├── EventHubEndpoints: map[string]Object (2 properties)
+│   │   │   │   ├── PartitionCount: *int
+│   │   │   │   └── RetentionTimeInDays: *int
+│   │   │   ├── Features: *Enum (2 values)
+│   │   │   │   ├── "DeviceManagement"
+│   │   │   │   └── "None"
+│   │   │   ├── IpFilterRules: Object (3 properties)[]
+│   │   │   │   ├── Action: *Enum (2 values)
+│   │   │   │   │   ├── "Accept"
+│   │   │   │   │   └── "Reject"
+│   │   │   │   ├── FilterName: *string
+│   │   │   │   └── IpMask: *string
+│   │   │   ├── MessagingEndpoints: map[string]Object (3 properties)
+│   │   │   │   ├── LockDurationAsIso8601: *string
+│   │   │   │   ├── MaxDeliveryCount: Validated<*int> (2 rules)
+│   │   │   │   │   ├── Rule 0: Maximum: 100
+│   │   │   │   │   └── Rule 1: Minimum: 1
+│   │   │   │   └── TtlAsIso8601: *string
+│   │   │   ├── MinTlsVersion: *string
+│   │   │   ├── NetworkRuleSets: *Object (3 properties)
+│   │   │   │   ├── ApplyToBuiltInEventHubEndpoint: *bool
+│   │   │   │   ├── DefaultAction: *Enum (2 values)
+│   │   │   │   │   ├── "Allow"
+│   │   │   │   │   └── "Deny"
+│   │   │   │   └── IpRules: Object (3 properties)[]
+│   │   │   │       ├── Action: *Enum (1 value)
+│   │   │   │       │   └── "Allow"
+│   │   │   │       ├── FilterName: *string
+│   │   │   │       └── IpMask: *string
+│   │   │   ├── PublicNetworkAccess: *Enum (2 values)
+│   │   │   │   ├── "Disabled"
+│   │   │   │   └── "Enabled"
+│   │   │   ├── RestrictOutboundNetworkAccess: *bool
+│   │   │   ├── Routing: *Object (4 properties)
+│   │   │   │   ├── Endpoints: *Object (4 properties)
+│   │   │   │   │   ├── EventHubs: Object (9 properties)[]
```

Comment:
If `ConnectionString` contains a secret (I bet it does), this should be a secret as well?

Same applies to all of the ConnectionStrings in this Routing `Endpoints` object. Possibly we can tell from example ARM templates that use these fields?